lib/broadway/message.ex

```diff
@@ -61,6 +63,22 @@ defmodule Broadway.Message do
     %Message{message | batch_key: batch_key}
   end
 
+  @doc """
+  Sets the batching mode for the message.
+
+  When the mode is `:bulk`, the batch that the message is in is delivered after
+  the batch size or the batch timeout is reached.
+
+  When the mode is `:flush`, the batch that the message is in is delivered
+  immediately.
+
+  The default mode for messages is `:bulk`.
+  """
+  @spec put_batching_mode(message :: Message.t(), mode :: :bulk | :flush) :: Message.t()
+  def put_batching_mode(%Message{} = message, mode) when mode in [:bulk, :flush] do
```

Comment:
maybe it should be called `put_batch_mode` so it is similar to `put_batch_key/2`?